### ISSUE TITLE
Move router state to mobx

### ISSUE THREAD

**Issue by webberwang** [codelab-app/platform#3140]
## Description

I see many hooks

![useAppQuery_hook_ts_—_platform](https://github.com/codelab-app/platform/assets/26264925/34465faa-8a5c-405f-88d4-066caf9db6f7)

but I think better to have a dedicated mobx service

![Cursor_and_router_service_ts_—_platform](https://github.com/codelab-app/platform/assets/26264925/fa4b9c21-4143-4020-961f-1b572ea35bae)

and will be initialized in _app.tsx

![_app_tsx_—_platform](https://github.com/codelab-app/platform/assets/26264925/b425bc03-e49d-4bba-8b30-c042d21f5bd8)

---

Basically we want to remove all router state hooks
